model_packaging/model_packagin.md

```diff
@@ -0,0 +1,155 @@
+     trained_model: (Required for trainable)
+       data_store: (Required) datastore for the training result source
+       bucket: (Required) Bucket that has the training result source
+       path: (Required) Bucket path that has the training result source
+       url: (Optional) Link to the model
+     trained_model_local: (Optional)
+       path: (Optional) Path to pull trained model in the user local machine
+   data_source: (Optional)
+     training_data: (Required for trainable)
+       data_store: (Required) datastore for the model data source
+       bucket: (Required) Bucket that has the model data source
+       path: (Required) Bucket path that has the model data source
+       url: (Optional) Link to the model
+     training_data_local: (Optional)
+       path: (Optional) Initial data files in the user local machine
+   mount_type: (Required) object storage mount type
+   evaluation_metrics: (optional) Define the metrics for the training job.
+     type: (Required) evaluation_metrics type
+     in: (Required) Path to store the evaluation_metrics
+   training_container_image: (Optional)
+     container_image_url: (Optional) Custom training container image url
+     container_store: (Optional) container_store for the custom training image
+   execution: (Required for trainable)
+     command: (Required) Entrypoint commands to execute model code
+       name: (Required) T-shirt size for training on Watson Machine Learning
+       nodes: (Required) Number of nodes needed for this training job. Default: 1
+   training_params: (Optional) list of hyperparameters for the training model
+   	- (optional) list of key(param name):value(param value)
+```
+
+## Data required for Model Serving
+
+```
+ serve: (Optional)
+   servable: (Optional) Indicate the model is servable without training. Default: False
+   tested_platforms (optional list): platform on which this model can served (current options: kubernetes, knative, seldon, wml, kfserving)
+   model_source: (Optional) - (Required if servable is true)
+     servable_model: (Required for s3 or url type)
+       data_store: (Required for s3 type) datastore for the model source
+       bucket: (Required for s3 type) Bucket that has the model source
+       path: (Required for s3 type) Source path to the model
+       url: (Required for url type) Source URL for the model
+     servable_model_local: (Optional)
+       path: (Optional) Servable model path in the user local machine
+   serving_container_image: (Required for container type)
+     container_image_url: (Required for container type) Container image to serve the model.
+     container_store: (Optional) container_store name
+```
+
+## Data Metedata 
+
+```
+ data (Optional)
+   source_id: (Optional) Extension file id regarding the data source.
+   domain: (Optional) Metadata about the data domain.
+   website: (Optional) Links to the data description
+   license: (Optional) Data license
+```
```

Comment:
metadata:
  # Optional section
    schema: 
    #Specify the features required for scoring with the model (input, type, shape)

      link_to_dataset: xxx
      # Linkage to model dataset / profile
      
      features: xxx
      # Captures model features
      
      service_schema: xxx
      # Service schema (generate Swagger client)
    output_schema: xxx
    # Output format, in contract (with same type of validation check as the inputs) (could explore compatible version ranges, etc.)
    metrics:
      key: value
      latency: 0.01
      # key/value pairs with user defined metrics (from training)
      # Model performance metrics (latency to inference)
    sampleInputs: http://uri/inputs.txt
    tags:
    # Arbitrary key / value pairs
    
    key: value 
    # Used to express anything not specified in the loose typing above